templates/addons/cluster-autoscaler/cluster-autoscaler.yaml

```diff
@@ -16,7 +16,7 @@ spec:
         app: ${CLUSTER_NAME}-cluster-autoscaler
     spec:
       containers:
-        - image: registry.k8s.io/autoscaling/cluster-autoscaler:${CLUSTER_AUTOSCALER_VERSION:=v9.37.0}
+        - image: registry.k8s.io/autoscaling/cluster-autoscaler:${KUBERNETES_VERSION%.*}.0
```

Comment:
```suggestion
        - image: registry.k8s.io/autoscaling/cluster-autoscaler:${CLUSTER_AUTOSCALER_VERSION:=${KUBERNETES_VERSION%.*}.0}
```
<sup>tinyyyyyyyy(🤏)</sup> nit: We may want to still allow users to set a specific `CLUSTER_AUTOSCALER_VERSION` and override the default.

Example:

```sh
$ clusterctl generate cluster test --kubernetes-version v1.30.1 --infrastructure local-linode:v0.0.0 --flavor kubeadm-cluster-autoscaler | yq 'select(.kind == "Deployment").spec.template.spec.containers[].image'
registry.k8s.io/autoscaling/cluster-autoscaler:v1.30.0

$ export CLUSTER_AUTOSCALER_VERSION=v1.30.1

$ clusterctl generate cluster test --kubernetes-version v1.30.1 --infrastructure local-linode:v0.0.0 --flavor kubeadm-cluster-autoscaler | yq 'select(.kind == "Deployment").spec.template.spec.containers[].image'
registry.k8s.io/autoscaling/cluster-autoscaler:v1.30.1
```